src/Adapter/MSTestAdapter.PlatformServices/Execution/TestMethodInfo.cs

```diff
@@ -394,12 +392,47 @@ private async Task<TestResult> ExecuteInternalAsync(object?[]? arguments, Cancel
         {
             try
             {
-                if (_classInstance != null && SetTestContext(_classInstance, result))
+                // We invoke global test initialize methods before creating the test class instance.
+                // We consider the test class constructor as a "local" initialization.
+                // We want to invoke first the global initializations, then local ones, then test method.
+                // After that, we invoke local cleanups (including Dispose) and finally global cleanups at last.
+                foreach ((MethodInfo method, TimeoutInfo? timeoutInfo) in Parent.Parent.GlobalTestInitializations)
+                {
+                    InvokeGlobalInitializeMethod(method, ref executionContext, timeoutInfo, timeoutTokenSource);
+                }
+
+                // TODO remove dry violation with TestMethodRunner
+                bool setTestContextSucessful = false;
+                if (executionContext is null)
```

Comment:
Please add a small comment (can be a follow-up) to explain here that the Invoke might change execution context so we can "understand" from here without drilling down.